source/extensions/omni.isaac.lab/omni/isaac/lab/managers/observation_manager.py

```diff
@@ -166,6 +171,11 @@ def reset(self, env_ids: Sequence[int] | None = None) -> dict[str, float]:
         for group_cfg in self._group_obs_class_term_cfgs.values():
             for term_cfg in group_cfg:
                 term_cfg.func.reset(env_ids=env_ids)
+                # reset modifiers if they are callable classes
+                if term_cfg.modifiers is not None:
+                    for mod in term_cfg.modifiers:
+                        if inspect.isclass(mod.func):
+                            mod.func.reset(env_ids=env_ids)
```

Comment:
There is a bug here. This won't be called for observation terms that are not classes.

Reply:
the ones that are not classes will not have anything to reset right?

Reply:
It iterates over: `_group_obs_class_term_cfgs` which are only observation terms that are classes

Reply:
I now made a separate list to keep track of modifiers that are classes. Should resolve the issue though we miss the test for it 😬